pytorch_lightning/metrics/functional/regression.py

```diff
@@ -280,16 +280,24 @@ def ssim(
 
     channel = pred.size(1)
     kernel = _gaussian_kernel(channel, kernel_size, sigma, device)
-    mu_pred = F.conv2d(pred, kernel, groups=channel)
-    mu_target = F.conv2d(target, kernel, groups=channel)
 
-    mu_pred_sq = mu_pred.pow(2)
-    mu_target_sq = mu_target.pow(2)
-    mu_pred_target = mu_pred * mu_target
-
-    sigma_pred_sq = F.conv2d(pred * pred, kernel, groups=channel) - mu_pred_sq
-    sigma_target_sq = F.conv2d(target * target, kernel, groups=channel) - mu_target_sq
-    sigma_pred_target = F.conv2d(pred * target, kernel, groups=channel) - mu_pred_target
+    # Concatenate
+    # pred for mu_pred
+    # target for mu_target
+    # pred * pred for sigma_pred
+    # target * target for sigma_target
+    # pred * target for sigma_pred_target
+    input_list = torch.cat([pred, target, pred * pred, target * target, pred * target])  # (5 * B, C, H, W)
+    outputs = F.conv2d(input_list, kernel, groups=channel)
+    output_list = [outputs[x * pred.size(0): (x + 1) * pred.size(0)] for x in range(len(outputs))]
+
+    mu_pred_sq = output_list[0].pow(2)
```

Comment:
is there a significant speed-up, otherwise, I feel to prefer previous readability

Reply:
I think it would be a little speed up as 1 convolution VS 5 convolutions

Reply:
yes, but the one convolution is on 5times larger input... correct?

Reply:
Yes, it's 5 times

Reply:
then I would personally stay with the longer but more talkative description 🐰 
cc: @justusschock

Reply:
@Borda If the speed is comparable yes, But maybe if you have one large cone that may be faster, since other algorithms may be selected to compute it. 

@ydcjeff can you benchmark this version against the old one?

Reply:
@ydcjeff sorry for this side path...